notes_on_refreshing_api.md

```diff
@@ -0,0 +1,29 @@
+# Towards BEMB `v1.0`.
+We are planning to refine and expand the current API of `BEMBFlex`.
+## The `pred_item` and multiple class prediction.
+- [ ] Currently the model supports predicting *binary* `batch.label` or multi-class `batch.item_index`. We plan to support arbitrary multi-class classifications.
+    * In particular, you don't need to change anything if `pred_item=True`, the model will know the number of classes is exactly the `num_items` parameter. Also, in this case, your `ChoiceDataset` object does **not** need to have a `label` attribute, since the model will look for the `item_index` as the ground truth for training.
+    * In contrast, if `pred_item=False`, now you need to supply a `num_classes` to the `BEMBFlex.__init__()` method. Also, you would need a `label` attribute in the `ChoiceDataset` object. The `label` attribute should be a `LongTensor` with values from `{0, 1, ..., num_classes}`.
```

Comment:
in more supermarkets, we have utils 0.0, 0.3, 0.5 and 1.0; should we allow float values here? I think we should. But that makes it complicated because that is then treating these as values with ordering rather than independent numbers.